pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTVariableDeclaratorId.java

```diff
@@ -142,9 +142,26 @@ public boolean isResourceDeclaration() {
      * since the type node is absent.
      */
     public boolean isTypeInferred() {
+        return isLambdaExpression() || isLocalVariableTypeInferred();
+    }
+
+    private boolean isLambdaExpression() {
         return jjtGetParent() instanceof ASTLambdaExpression;
     }
 
+    private boolean isLocalVariableTypeInferred() {
+        boolean hasType = true;
+
+        if (jjtGetParent() instanceof ASTResource) {
+            // covers "var" in try-with-resources
+            hasType = jjtGetParent().getFirstChildOfType(ASTType.class) != null;
```

Comment:
We could remove the local variable by returning in the `if`s, ie  `return ... == null`